lib/translator/commands/array_stencil_command.rb

```diff
@@ -18,6 +97,19 @@ def visit_array_stencil_command(command)
                 # of the command in the environment.
                 env_builder = @environment_builder[command.unique_id]
 
+                # Translate relative indices to 1D-indicies starting by 0
+                if command.use_parameter_array
+                    offsets_mapped = Hash.new
```

Comment:
Can we make `ArrayStencilCommand.offsets` always an array of arrays? I.e., when running the constructor, checking if its an array of arrays (such that the API does not have to change). If not, then do the conversion there instead of here.